apps/04TowersOfHanoi.js

```diff
@@ -7,36 +7,79 @@ var rl = readline.createInterface({
   output: process.stdout
 });
 
+
 var stacks = {
   a: [4, 3, 2, 1],
   b: [],
   c: []
 };
 
+var moves = 1;
+
 function printStacks() {
   console.log("a: " + stacks.a);
   console.log("b: " + stacks.b);
   console.log("c: " + stacks.c);
 }
 
-function movePiece() {
-  // Your code here
-
+function movePiece(startStack, endStack) {
+  //Take peice 
+  var piece = stacks[startStack].pop();
+  //Place piece
+  stacks[endStack].push(piece);
+  
+  
+  
 }
 
-function isLegal() {
-  // Your code here
-
+function isLegal(startStack, endStack) {
+  var startPiece = (stacks[startStack][(stacks[startStack].length-1)]);
+  var endPiece = (stacks[endStack][(stacks[endStack].length-1)]);
+ // if array is empty it will return undefined
+   if  (endPiece === undefined) {
```

Comment:
We need to stay in the mindset of being detail oriented.  Make sure all code within one level of curly braces is all aligned.